x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/PreparedStatementTestCase.java

```diff
@@ -341,50 +412,63 @@ public void testSingleParameterMultipleTypes() throws SQLException {
                 statement.setInt(1, intVal);
-                assertEquals(new Tuple<>(JDBCType.INTEGER.getVendorTypeNumber(), intVal), execute(statement));
+                assertEquals(new Tuple<>(INTEGER.getName(), intVal), execute(statement));
                 statement.setLong(1, longVal);
-                assertEquals(new Tuple<>(JDBCType.BIGINT.getVendorTypeNumber(), longVal), execute(statement));
+                assertEquals(new Tuple<>(LONG.getName(), longVal), execute(statement));
                 statement.setFloat(1, floatVal);
-                assertEquals(new Tuple<>(JDBCType.REAL.getVendorTypeNumber(), floatVal), execute(statement));
+                assertEquals(new Tuple<>(FLOAT.getName(), floatVal), execute(statement));
                 statement.setDouble(1, doubleVal);
-                assertEquals(new Tuple<>(JDBCType.DOUBLE.getVendorTypeNumber(), doubleVal), execute(statement));
+                assertEquals(new Tuple<>(DOUBLE.getName(), doubleVal), execute(statement));
                 statement.setNull(1, JDBCType.DOUBLE.getVendorTypeNumber());
-                assertEquals(new Tuple<>(JDBCType.DOUBLE.getVendorTypeNumber(), null), execute(statement));
+                assertEquals(new Tuple<>(DOUBLE.getName(), null), execute(statement));
                 statement.setBoolean(1, booleanVal);
-                assertEquals(new Tuple<>(JDBCType.BOOLEAN.getVendorTypeNumber(), booleanVal), execute(statement));
+                assertEquals(new Tuple<>(BOOLEAN.getName(), booleanVal), execute(statement));
                 statement.setByte(1, byteVal);
-                assertEquals(new Tuple<>(JDBCType.TINYINT.getVendorTypeNumber(), byteVal), execute(statement));
+                assertEquals(new Tuple<>(BYTE.getName(), byteVal), execute(statement));
                 statement.setShort(1, shortVal);
-                assertEquals(new Tuple<>(JDBCType.SMALLINT.getVendorTypeNumber(), shortVal), execute(statement));
+                assertEquals(new Tuple<>(SHORT.getName(), shortVal), execute(statement));
             }
         }
     }
 
-    private Tuple<Integer, Object> execute(PreparedStatement statement) throws SQLException {
+    private Tuple<String, Object> execute(PreparedStatement statement) throws SQLException {
         try (ResultSet results = statement.executeQuery()) {
             ResultSetMetaData resultSetMetaData = results.getMetaData();
             assertTrue(results.next());
-            Tuple<Integer, Object> result = new Tuple<>(resultSetMetaData.getColumnType(1), results.getObject(1));
+            Tuple<String, Object> result = new Tuple<>(resultSetMetaData.getColumnTypeName(1), results.getObject(1));
             assertFalse(results.next());
             return result;
         }
     }
 
-    private static long testMillis(long randomMillis, int i) {
+    private static long adjustDateForEachDocument(long randomMillis, int i) {
```

Comment:
You pass a random time to the method, and then adjust the day based on the index `i` so that each document has a slightly different date, so that the `WHERE` filtering with the JDBC timestamp param is tested properly.